Jenkinsfile

```diff
@@ -847,17 +847,62 @@ def withBeatsEnv(boolean archive, Closure body) {
         }
       } finally {
         if (archive) {
-          catchError(buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
-            junitAndStore(allowEmptyResults: true, keepLongStdio: true, testResults: "**/build/TEST*.xml")
-            archiveArtifacts(allowEmptyArchive: true, artifacts: '**/build/TEST*.out')
-          }
+          archiveTestOutput(testResults: '**/build/TEST*.xml', artifacts: '**/build/TEST*.out')
         }
         reportCoverage()
       }
     }
   }
 }
 
+/**
+  This method archives and report the tests output.
+    It searches for certain folders to bypass some issues when working with big repositories. In addition,
+
+    OUTPUT:
+      - JUnit
+      - Artifacts in Jenkins
+      - Artifacts in GCP
+*/
+def archiveTestOutput(Map args = [:]) {
+  catchError(buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
+    if (isUnix()) {
+      fixPermissions("${WORKSPACE}")
+    }
+    shOrBat(label: 'Prepare test output', script: 'python .ci/scripts/pre_archive_test.py')
```

Comment:
Python could be an alternative to Shell to be cross-platform, right?

Reply:
I don't see any python specific step, what do you mean?

Reply:
Sorry, maybe I did not expresses myself correctly: I meant that the `shOrbat` step is great to invoke a script in a cross-platform manner. Then the target script must be cross-platform too, and python seems a good candidate. In the above block, the `pre_archive_test.py`.

Writing scripts in python would give us that power of cross-platform-_tivity_